framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -586,10 +586,18 @@ void MediaPlayerImpl::notifyObserver(player_observer_command_t cmd, ...)
 
 void MediaPlayerImpl::playback()
 {
-	ssize_t num_read = mInputHandler.read(mBuffer, (int)mBufSize);
+	ssize_t num_read = 0;
+	auto source = mInputHandler.getInputDataSource();
+	int channels = source->getChannels();
+	
+	if (channels == AUDIO_OUTPUT_CHANNELS) {
```

Comment:
Please refer to my old comment what I gave to you before I left.
This is wrong. This will bring critical error if song made of 6 channels or so on...
Player or Recorder should not care about lower layer's capacity.